packages/types/step/index.ts

```diff
@@ -0,0 +1,28 @@
+import {EventEmitter} from 'events';
+type StepEventHandler<T extends Step> = (e: StepEventData<T>) => any;
+type StepStatus = 'pending' | 'in progress' | 'done '; // can have more here
+export interface StepEventData<T extends Step> {
+  step: T;
+  date: Date;
+}
+interface Step extends EventEmitter {
```

Comment:
It might be unpopular opinion but since we use interfaces and object oriented language events may not really match this idea and may be another source for ugly workarounds. 
Do you think we can model this without events?